okhttp/src/main/java/okhttp3/internal/tls/TrustRootIndex.kt

```diff
@@ -13,11 +13,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package okhttp3.internal.tls;
+package okhttp3.internal.tls
 
-import java.security.cert.X509Certificate;
+import java.security.cert.X509Certificate
 
-public interface TrustRootIndex {
-  /** Returns the trusted CA certificate that signed {@code cert}. */
-  X509Certificate findByIssuerAndSignature(X509Certificate cert);
+interface TrustRootIndex {
+  /** Returns the trusted CA certificate that signed [cert].  */
+  fun findByIssuerAndSignature(cert: X509Certificate): X509Certificate
```

Comment:
Build will currently fail. This returns a non-null `X509Certificate` but `AndroidPlatform` returns null when overriding this method.

Reply:
Any ideas on how to proceed here? Should this be nullable because `AndroidPlatform` overrides this and returns null?

Reply:
Should return `X509Certificate?`.

Classes in `internal` are not annotated `@Nullable` precisely!

Reply:
Thanks Jesse, good to know! 
92433df501d10342a66c242a114fdee52aa33218